spec/src/modules/catalog.js

```diff
@@ -3120,4 +3135,453 @@ describe('ConstructorIO - Catalog', () => {
+
+    before(async () => {
+      const { catalog } = new ConstructorIO({
+        ...validOptions,
+        fetch: fetchSpy,
+      });
+
+      await catalog.addFacetConfiguration(facetConfiguration);
+    });
+
+    after(async () => {
+      const { catalog } = new ConstructorIO({
+        ...validOptions,
+        fetch: fetchSpy,
+      });
+
+      // Clean up all the facet/facet option configurations that were created
+      await catalog.removeFacetConfiguration({ name: facetConfiguration.name });
+    });
+
+    describe('addFacetOptionConfiguration', () => {
+      let mockFacetOptionConfiguration = createMockFacetOptionConfiguration(facetGroupName);
+
+      it('Should resolve when adding a facet option configuration', (done) => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        catalog.addFacetOptionConfiguration(mockFacetOptionConfiguration).then(() => done());
+      });
+
+      it('Should return error when adding a facet option configuration that already exists', () => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        return expect(catalog.addFacetOptionConfiguration(mockFacetOptionConfiguration)).to.eventually.be.rejected;
+      });
+
+      it('Should return error when adding a facet option configuration with unsupported options', () => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        mockFacetOptionConfiguration = createMockFacetOptionConfiguration(facetGroupName);
+        mockFacetOptionConfiguration.type = 'hidden';
+
+        return expect(catalog.addFacetOptionConfiguration(mockFacetOptionConfiguration)).to.eventually.be.rejected;
+      });
+
+      it('Should return error when adding a facet option configuration with unsupported option values', () => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        mockFacetOptionConfiguration = createMockFacetOptionConfiguration(facetGroupName);
+        mockFacetOptionConfiguration.position = 'one';
+
+        return expect(catalog.addFacetOptionConfiguration(mockFacetOptionConfiguration)).to.eventually.be.rejected;
+      });
+    });
+
+    describe('addOrModifyFacetOptionConfigurations', () => {
+      let mockFacetOptionConfigurations = [
+        createMockFacetOptionConfiguration(),
+        createMockFacetOptionConfiguration(),
+      ];
+
+      it('Should resolve when adding facet configurations', (done) => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        catalog.addOrModifyFacetOptionConfigurations({
+          facetGroupName,
+          facetOptionConfigurations: mockFacetOptionConfigurations,
+        }).then(() => done());
```

Comment:
I think this `.then(() => done());` can be `.then(done);`

Reply:
Jus adding a comment here for others. `done` isn't expecting any arguments to be passed to it and errors if there is. So we have to leave it has `.then(() => done())`.